fms_fsdp/utils/train_utils.py

```diff
@@ -29,29 +29,48 @@ def train(
     start_step,
     tokens_seen,
 ):
-    if cfg.use_wandb:
-        try:
-            import wandb
-        except ImportError:
-            raise ImportError(
-                "use_wandb is set to True but wandb is not installed. Please install wandb to use wandb support."
-            )
+    if cfg.tracker:
+        if cfg.tracker not in ["wandb", "aim"]:
+            raise ValueError(f"tracker {cfg.tracker} not supported.")
+        tracker_dir = cfg.tracker_dir
+        project_name = cfg.tracker_project_name
+        run_id = cfg.tracker_run_id
+        hparams = {
+            "learning_rate": cfg.learning_rate,
+            "num_steps": cfg.num_steps,
+            "batch_size": cfg.batch_size,
+        }
+
+        if cfg.tracker == "wandb":
+            try:
+                import wandb  # type: ignore
+            except ImportError:
+                raise ImportError("tracker is set to wandb but wandb is not installed.")
+            if rank == 0:
+                print(
+                    f"--> wandb is enabled! Make sure to pass your wandb api key via WANDB_API_KEY."
```

Comment:
nit: this looks a bit odd to me, I'd expect to only get the warning about needing to pass the API KEY if it failed to connect

Reply:
If it is not passed, wandb will try to pop an interactive thing to ask you fill username, password, etc. which might be a little confusing for some users. Adding this language is to simply add a hint on "what you might miss to do" or "oh, this is how I should pass credentials".
But I am also fine with removing this "reminder" given people who choose to explicitly turn on wandb might be very well faimilar with all these.

Reply:
addressed in https://github.com/foundation-model-stack/fms-fsdp/pull/55/commits/4df07985ebf1e7ed9a04809fb8e3221291092132